### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by sadaf-Daneshgar**
**New Features**
-----------
- 🔰 Add unit tests for all the methods in all the classes:
  - `Book`
  - `Person`
  - `Student`
  - `Teacher`
  - `Classroom`
  - `Rental`
  - 
👥 Authors
----------------
@sadaf-Daneshgar 
@MohammadYaser

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  it 'returns correct name' do
+    expect(person.correct_name).to eq('Ali')
+  end
+
+  it 'adds rental' do
+    book = double('book')
+    expect(Rental).to receive(:new).with(any_args).and_return(double('rental'))
+    person.add_rental('2023-01-01', book)
+  end
```
Comment (mckent05) [inline spec/person_spec.rb:51]: - You have done well so far, please complete this test after calling the `add_book` method to ensure that the rentals array contains the rental added. Please fix this 🙏

File: spec/trimmer_decorator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  end
+
+  it 'returns the correct name using the decorator' do
+    expect(decorator_instance.correct_name).to eq('Ali')
+  end
+
+  it 'updates the nameable attribute' do
+    new_nameable_instance = TestNameable.new
+    decorator_instance.nameable = new_nameable_instance
+    expect(decorator_instance.correct_name).to eq('Ali')
```
Comment (mckent05) [inline spec/trimmer_decorator_spec.rb]: - You have done well so far, but please write tests to show that your trimmer decorator works as expected and gets the first 10 letters of the name when the name has more than 10 characters. Please fix this 👍 👍

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @sadaf-Daneshgar, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- You did well testing all classes and decorators 👍 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — APPROVED** [review]
# APPROVED 🟢 ✔️ 

## Hi team👋, 

Great job implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ All tests are passing ✔️ 
✔️ No linter errors 🟢 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**